pkg/tagger/collectors/kubernetes_main.go

```diff
@@ -0,0 +1,119 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubeapiserver
+
+package collectors
+
+import (
+	"time"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+)
+
+const (
+	kubernetesCollectorName = "kubernetes"
+)
+
+type KubernetesCollector struct {
+	kubeUtil  *kubelet.KubeUtil
+	apiClient *apiserver.APIClient
+	infoOut   chan<- []*TagInfo
+
+	// used to set a custom delay
+	lastUpdate time.Time
+	updateFreq time.Duration
+}
+
+// Detect tries to connect to the kubelet
+func (c *KubernetesCollector) Detect(out chan<- []*TagInfo) (CollectionMode, error) {
+	if config.Datadog.GetBool("kubernetes_collect_service_tags") == false {
+		return NoCollection, nil
```

Comment:
You can return a custom error "collection disabled by the configuration", this will show up as a `debug` tagger.go logline